dvc/serialize.py

```diff
@@ -26,7 +32,7 @@ def _get_outs(stage: "PipelineStage"):
             bucket_key += ["no_cache"]
         key = "_".join(bucket_key)
         outs_bucket[key] = outs_bucket.get(key, []) + [o.def_path]
-    return outs_bucket
+    return [(key, outs_bucket[key]) for key in outs_bucket.keys()]
```

Comment:
Notice that we do not sort for `_get_outs()` at all. This will likely make dump for pipeline file non-deterministic. Not really
focusing it for now, as we will be rarely dumping to the pipeline file.